toolchain/lexer/token_kind.h

```diff
@@ -87,21 +71,13 @@ class TokenKind {
     return false;
   }
 
-  // Enable conversion to our private enum, including in a `constexpr` context,
-  // to enable usage in `switch` and `case`. The enum remains private and
-  // nothing else should be using this.
-  // NOLINTNEXTLINE(google-explicit-constructor)
-  constexpr operator KindEnum() const { return kind_value_; }
-
   // Prints the TokenKind, typically for diagnostics.
```

Comment:
Comment I added in my version of this PR (hadn't mailed it out as it wasn't clear if the review for the baseline would need changes):
```suggestion
  // Override the EnumBase printing to use the fixed spelling rather than the
  // name for tokens as this better corresponds to the source code the
  // represent.
```